core/unix/signal_linux_riscv64.c

```diff
@@ -121,8 +126,56 @@ signal_frame_extra_size(bool include_alignment)
     return 0;
 }
 
+static void
+catch_sigill(int signo, kernel_siginfo_t *si, void *data)
+{
+    (void)signo;
+    (void)si;
+    (void)data;
+
+    sigill_caught = 1;
+    dr_longjmp(&jmpbuf, 1);
+}
+
+static int
+sigill_detected(void *func)
+{
+    sigill_caught = 0;
+
+    kernel_sigaction_t act = { 0 };
+    kernel_sigaction_t old_act;
+    act.flags = SA_ONSTACK | SA_RESTART | SA_SIGINFO;
+    act.handler = catch_sigill;
```

Comment:
Seems safest to use set_handler_sigact(), like the x86 SIGILL at init does, rather than replicating flags here and now having more places to update/maintain.